src/common/column_matrix.h

```diff
@@ -167,6 +171,7 @@ class ColumnMatrix {
    */
   void InitFromSparse(SparsePage const& page, const GHistIndexMatrix& gmat, double sparse_threshold,
                       int32_t n_threads) {
+    is_initialized_ = true;
```

Comment:
We have a few places for initialization. From adapter, from ghist index matrix, and from external memory.